### ISSUE TITLE
No uvloop in docker image

### ISSUE THREAD

**Issue by TobiasStockmanns** [bachya/ecowitt2mqtt#963]
**Describe the bug**
When running the docker image `bachya/ecowitt2mqtt:latest` the following error message is displayed:
```
docker run -it bachya/ecowitt2mqtt:latest
Traceback (most recent call last):
  File "/venv/bin/ecowitt2mqtt", line 5, in <module>
    from ecowitt2mqtt.__main__ import main
  File "/venv/lib/python3.11/site-packages/ecowitt2mqtt/__main__.py", line 10, in <module>
    import uvloop
  File "/venv/lib/python3.11/site-packages/uvloop/__init__.py", line 9, in <module>
    from .loop import Loop as __BaseLoop  # NOQA
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ModuleNotFoundError: No module named 'uvloop.loop'
```

**To Reproduce**
Run on a Raspberry Pi `docker run -it bachya/ecowitt2mqtt`

**Expected behavior**
Clean start of container

**Comment by bachya** [comment]
What Pi version and CPU architecture are you using?

**Comment by TobiasStockmanns** [comment]
Release = Raspbian GNU/Linux 11 (bullseye)
Kernel = Linux 6.1.21-v8+
Platform = Raspberry Pi 4 Model B Rev 1.2
CPU Architecture = BCM2711B0 / C0

I hope this information helps.

**Comment by bachya** [comment]
What does this command output?

```
cat /proc/cpuinfo
```

**Comment by andreaalloway** [comment]
Also seeing the same bug on the Raspberry Pi 4

`` cat /proc/cpuinfo
processor       : 0
model name      : ARMv7 Processor rev 3 (v7l)
BogoMIPS        : 126.00
Features        : half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt                                                                                                                                                                                                                                              vfpd32 lpae evtstrm crc32
CPU implementer : 0x41
CPU architecture: 7
CPU variant     : 0x0
CPU part        : 0xd08
CPU revision    : 3

processor       : 1
model name      : ARMv7 Processor rev 3 (v7l)
BogoMIPS        : 126.00
Features        : half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt                                                                                                                                                                                                                                              vfpd32 lpae evtstrm crc32
CPU implementer : 0x41
CPU architecture: 7
CPU variant     : 0x0
CPU part        : 0xd08
CPU revision    : 3

processor       : 2
model name      : ARMv7 Processor rev 3 (v7l)
BogoMIPS        : 126.00
Features        : half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt                                                                                                                                                                                                                                              vfpd32 lpae evtstrm crc32
CPU implementer : 0x41
CPU architecture: 7
CPU variant     : 0x0
CPU part        : 0xd08
CPU revision    : 3

processor       : 3
model name      : ARMv7 Processor rev 3 (v7l)
BogoMIPS        : 126.00
Features        : half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt                                                                                                                                                                                                                                              vfpd32 lpae evtstrm crc32
CPU implementer : 0x41
CPU architecture: 7
CPU variant     : 0x0
CPU part        : 0xd08
CPU revision    : 3

Hardware        : BCM2711
Revision        : c03112
Serial          : 1000000057f96e17
Model           : Raspberry Pi 4 Model B Rev 1.2

**Comment by TobiasStockmanns** [comment]
I get the following information from `cpuinfo`:
```
processor	: 0
BogoMIPS	: 108.00
Features	: fp asimd evtstrm crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x0
CPU part	: 0xd08
CPU revision	: 3

processor	: 1
BogoMIPS	: 108.00
Features	: fp asimd evtstrm crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x0
CPU part	: 0xd08
CPU revision	: 3

processor	: 2
BogoMIPS	: 108.00
Features	: fp asimd evtstrm crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x0
CPU part	: 0xd08
CPU revision	: 3

processor	: 3
BogoMIPS	: 108.00
Features	: fp asimd evtstrm crc32 cpuid
CPU implementer	: 0x41
CPU architecture: 8
CPU variant	: 0x0
CPU part	: 0xd08
CPU revision	: 3

Hardware	: BCM2835
Revision	: c03112
Serial		: 100000007b6f2b4b
Model		: Raspberry Pi 4 Model B Rev 1.2
```

**Comment by bachya** [comment]
I do not have a Raspberry Pi 4 with which I can test, unfortunately. What happens in you manually install `uvloop`?

```
pip install uvloop
```

**Comment by Rob-2** [comment]
Hello bachya,

I have the same problem with the Homeassitant Addon on an Raspberry Pi 4 on bookworm.

I tried to run the container with:
> docker run --entrypoint /bin/bash -it ghcr.io/bachya/home-assistant-addon-ecowitt2mqtt-armv7:2024.06.2
and then a
> pip install uvloop

Result was:
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Requirement already satisfied: uvloop in /usr/lib/python3.11/site-packages (0.19.0)
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager, possibly rendering your system unusable.It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv. Use the --root-user-action option if you know what you are doing and want to suppress this warning.

Can I help debugging further?

**Comment by Rob-2** [comment]
OK, problem solved.

I have reinstalled HAOS on another Raspi4 and ecowitt2mqtt was starting.

I then cleaned the drive and installed Debian12-64 and then afterwards "HomeAssistant Supervised on Debian" according to:
https://community.home-assistant.io/t/installing-home-assistant-supervised-on-a-raspberry-pi-using-debian-12/
and ecowitt2mqtt was also starting.

So it seems the cause of the problem was somewhere in the upgrade and/or the use of Raspberry Pi OS, even though it's a Debian derivat.
